### PR TITLE
Added options to connect CLI to PyCharm remote debug server

### PR DESCRIPTION
**Pull request by sfc-gh-pjob**
### Pre-review checklist
   * [x] I've confirmed that instructions included in README.md are still correct after my changes in the codebase.
   * [ ] I've added new automated tests to verify correctness of my new code.
   * [x] I've confirmed that my changes are up-to-date with the target branch.
   * [x] I've described my changes in the section below.

### Changes description
Added options to connect CLI to PyCharm remote debug server. Details in `CONTRIBUTING.md`.

### CODE REVIEW

File: src/snowcli/pycharm_remote_debug.py [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+):
+    if pycharm_debug_library_path:
+        if (
+            pycharm_debug_server_host is not None
+            and pycharm_debug_server_port is not None
+        ):
+            import sys
+
+            sys.path.append(pycharm_debug_library_path)
+            import pydevd_pycharm
```
Comment (sfc-gh-turbaszek) [inline src/snowcli/pycharm_remote_debug.py:19]: Should this be added as dev dependcy to pyproject.toml?

Comment (sfc-gh-pjob) [reply src/snowcli/pycharm_remote_debug.py:19]: JetBrains instructions say that this library is versioned together with PyCharm IDE and that the versions have to match.

Line from their instruction:
```
pip install pydevd-pycharm~=<version of PyCharm on the local machine>
```

So I don't think it's possible to add it to pyproject.toml.

Comment (sfc-gh-turbaszek) [reply src/snowcli/pycharm_remote_debug.py:19]: Ok, got it. I assume this `pip install` is in the link, right?

**Review by sfc-gh-turbaszek — APPROVED** [review]

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -30,3 +30,33 @@ Every integration test should have `integration` mark. By default, integration t
...
+1. If the CLI and the IDE are on the same machine, you have to just copy the path to the file.
+1. If the CLI is on another machine, then you have to copy the file there and also copy the target file path.
+1. Run the CLI using `snow --pycharm-debug-library-path <path-to-pydevd-pycharm.egg> <other-options-and-command>`.
+    * Example: `snow --pycharm-debug-library-path "/Users/xyz/Library/Application Support/JetBrains/Toolbox/apps/IDEA-U/ch-0/231.9011.34/IntelliJ IDEA.app.plugins/python/debugger-eggs-output/pydevd-pycharm.egg" snowpark function list`
+    * The CLI will try to connect to your debug server (by default to `localhost:12345`).
+    * If a connection cannot be established, you will see some exception about it.
+    * If you want to use other host or port you can use `--pycharm-debug-server-host` and `--pycharm-debug-server-port` options.
+    * The code execution will be paused before execution of your command.
+      You will see the exact line in the IDE's debug view.
+      You can resume code execution, add breakpoints, evaluate variables, do all the things you usually do when debugging locally.
```
Comment (sfc-gh-astus) [inline CONTRIBUTING.md:62]: What happened to the numbering?

**Review by sfc-gh-astus — APPROVED** [review]

File: CONTRIBUTING.md (hunk 2 above)
Comment (sfc-gh-pjob) [reply CONTRIBUTING.md:62]: Nothing :) IMO it looks ok: https://github.com/Snowflake-Labs/snowcli/blob/main/CONTRIBUTING.md

**Pull request merged**